demes/__main__.py

```diff
@@ -0,0 +1,135 @@
+import sys
+import functools
+
+import click
+import ruamel
+
+import demes
+
+
+@click.group()
+def main():
+    pass
+
+
+@main.command(context_settings={"ignore_unknown_options": True})
+@click.option("-N0", "N0", type=float, required=True, help="Reference population size")
+@click.argument("commands", required=True, nargs=-1)
+def ms(N0, commands):
+    """
+    Convert an ms command to a Demes model. E.g. (from the ms manual):
+
+        demes ms 15 1000 -t 6.4 -G 6.93 -eG 0.2 0.0 -eN 0.3 0.5 -N0 20000
+    """
+    graph = demes.from_ms(" ".join(commands), N0=N0)
+    demes.dump(graph, sys.stdout)
+
+
+def detect_input_format(input_format, filename):
+    if input_format is None:
+        if filename.endswith("json"):
+            input_format = "json"
+        else:
+            input_format = "yaml"
+    return input_format
+
+
+@main.command()
+@click.option(
+    "-f",
+    "--fully-resolved",
+    is_flag=True,
+    default=False,
+    help="Output fully-resolved model. A simplified model is output by default.",
+)
+@click.option("-i", "--input-format", type=click.Choice(["yaml", "json"]), default=None)
+@click.option(
+    "-o", "--output-format", type=click.Choice(["yaml", "json"]), default="yaml"
+)
+@click.argument("filename", type=str, default="-")
```

Comment:
Better to use the built-in file support: https://click.palletsprojects.com/en/8.0.x/arguments/#file-arguments